homeassistant/components/vizio/config_flow.py

```diff
@@ -231,7 +263,98 @@ def __init__(self) -> None:
             discovery_info[CONF_HOST]
         )
 
-        # Form must be shown after discovery so user can confirm/update configuration before ConfigEntry creation.
+        # Form must be shown after discovery so user can confirm/update configuration
+        # before ConfigEntry creation.
         self._must_show_form = True
-
         return await self.async_step_user(user_input=discovery_info)
+
+    async def async_step_pair_tv(
+        self, user_input: Dict[str, Any] = None
+    ) -> Dict[str, Any]:
+        """Start pairing process and ask user for PIN to complete pairing process."""
+        errors = {}
+
+        # Start pairing process if it hasn't already started
+        if not self._ch_type and not self._pairing_token:
+            dev = VizioAsync(
```

Comment:
Do you handle if host is incorrect ?

Reply:
That is handled here: https://github.com/raman325/home-assistant/blob/81b31a806f96e8b3fae886a6a05478684ca6e321/homeassistant/components/vizio/config_flow.py#L294-L297
This sends the user back to the user step if the ch_type and pairing_token can't be retrieved, which could be because of an invalid host